merlin/models/tf/models/base.py

```diff
@@ -328,48 +378,37 @@ def fit(
         train_metrics_steps=1,
         **kwargs,
     ):
-        # Check if merlin-dataset is passed
-        if hasattr(x, "to_ddf"):
-            if not batch_size:
-                raise ValueError("batch_size must be specified when using merlin-dataset.")
-            from merlin.models.tf.dataset import BatchedDataset
-
-            x = BatchedDataset(x, batch_size=batch_size, **kwargs)
+        x = _maybe_convert_merlin_dataset(x, batch_size, **kwargs)
+        validation_data = _maybe_convert_merlin_dataset(
+            validation_data, batch_size, shuffle=False, **kwargs
+        )
+        callbacks = self._add_metrics_callback(callbacks, train_metrics_steps)
 
-        if hasattr(validation_data, "to_ddf"):
-            if not batch_size:
-                raise ValueError("batch_size must be specified when using merlin-dataset.")
-            from merlin.models.tf.dataset import BatchedDataset
+        fit_kwargs = {
+            k: v
+            for k, v in locals().items()
+            if k not in ["self", "kwargs", "train_metrics_steps", "__class__"]
+        }
 
-            kwargs.pop("shuffle", None)
-            validation_data = BatchedDataset(
-                validation_data, batch_size=batch_size, shuffle=False, **kwargs
-            )
+        return super().fit(**fit_kwargs)
 
-        callbacks = self._add_metrics_callback(callbacks, train_metrics_steps)
-
-        return super().fit(
-            x,
-            y,
-            batch_size,
-            epochs,
-            verbose,
-            callbacks,
-            validation_split,
-            validation_data,
-            shuffle,
-            class_weight,
-            sample_weight,
-            initial_epoch,
-            steps_per_epoch,
-            validation_steps,
-            validation_batch_size,
-            validation_freq,
-            max_queue_size,
-            workers,
-            use_multiprocessing,
+    def compute_loss_metrics(
+        self, inputs, targets, training: bool = False, compute_metrics=True, **kwargs
+    ):
+        predictions = self(inputs, training=training, **kwargs)
```

Comment:
When I run `model.fit()` setting `validation_data` argument, it will call `model.evaluate()` in the end of each epoch. Then `RetrievalModel.compute_loss_metrics()` is called and sets [here](https://github.com/NVIDIA-Merlin/models/blob/tf/retrieval-eval/merlin/models/tf/models/base.py#L596) `kwargs["eval_sampling"] = True`, so that in the `ItemRetrievalScorer.call()` the `eval_sampling` is `True`.
But the `eval_sampling`  is not being passed through for some reason, which causes an error in `ItemRetrievalScorer` when using `model.fit(..., validation_data=...)`